### ISSUE TITLE
[Feature] [Question] UI issues: strange navigation on article click & forced autoload

### ISSUE THREAD

**Issue by rg-software** [marienfressinaud/FreshRSS#4069]
I apologize for such a vague title. There are several very annoying UI decisions in an otherwise an excellent system, so I wanted to ask whether this behavior is intended or "to be improved". Here are the two I face most often.

1) When I read an article and want to move to the next one, the system navigates to some counter-intuitive place, in most cases closer to the end or even past the end of the article I am going to read. This can be seen in the gif below:

![Animation2](https://user-images.githubusercontent.com/13718460/146348132-bb6fddfc-ff53-4189-ae0e-4d1a3538cb96.gif)

2) Suppose I have a large batch of unread articles (~300). I can quickly scan the titles of the first page of the batch, maybe open and read some of them. Then I want to mark all these articles as "read" to remove them and proceed to the next batch. This is, for example, how Inoreader works. In FreshRSS I have to click on every single title to mark it as "read" as I go. If I approach the end of the page, the system loads the next unread articles, so I am forced to process them all in one sitting. In my view, this is unfortunate. I'd prefer to be able to have a button to mark all the articles of the page as "read" and then display the next page, not to proceed further automatically.

**Comment by Frenzie** [comment]
1. Which browser, theme, FreshRSS version, etc.? That's obviously not how it's supposed to work. ;-)
2. Alt+r for previous, Shift+r for all. (Or whatever you use instead of r.)

**Comment by rg-software** [comment]
Right...
1) Version 1.18.1, theme Swage (but I checked Default, same thing), Vivaldi browser (Chrome-based).
2) Alt+r is pretty close, thanks! So it means I have to click somewhere in the middle of the list and press it to mark everything above as read.

**Comment by Frenzie** [comment]
Which extensions do you use, whether in FreshRSS or Vivaldi? Because you pretty much just described my main setup (Vivaldi and Firefox are my primary browsers, and my regular instance is normally running the latest stable, in this case 1.18.1.)

The quickest way to test for a browser is probably `vivaldi --profile-directory /tmp` (or just regular Chromium if you haven't personalized that).

**Comment by rg-software** [comment]
Well, running the browser without extensions doesn't change anything (and maybe the quickest way to check it is just to open a private tab). I will try some other setups tomorrow then. Currently it works fine on my mobile device, but not on the desktop. BTW, no way to do "Alt + r" on mobile, right?..

**Comment by math-GH** [comment]
Could you please check your settings?
![grafik](https://user-images.githubusercontent.com/1645099/146427339-9008fde6-fa17-4c44-9701-afb595eb19f5.png)

First checkbox: if enabled, it should do what you need in your 1st point.
Seconde checkbox: if disabled, it would stop the auto load of next articles. Than you can mark the articles with "Mark as read" at the top of your page (see menu bar)

**Comment by Frenzie** [comment]
> and maybe the quickest way to check it is just to open a private tab

Only iff you don't allow any extensions in your private tabs. By "quickest" I meant quickest way to be actually 100% sure.

It would also rule out non-extension browser settings.

**Comment by rg-software** [comment]
@math-GH 
Oh, I really don't want to sound like I am hard to please, but this cure isn't much better than the disease...
Option 1 makes the whole UI... jumpy? When I open an article, I am sort of expecting it to be opened righ here, so I can close it back by clicking the mouse at the same place. To me it just feels natural.
As of Option 2, I tried pressing the "mark as read" button in the toolbar, and it marked as read all my 1500+ unread articles! It sort of solved my issue for today, but not sure I am happy :)

**Comment by math-GH** [comment]
I am sorry.

Can you reproduce the behavior of (1) also in demo.freshrss.org/ ?

**Comment by math-GH** [comment]
Checked with Vivaldi (5.0.2497.32) on Windows 10 on demo.fresh.org. Cannot reproduce your issue 1.

**Comment by rg-software** [comment]
@math-GH Sorry, still need to look a bit further. But isn't it true that the "Demo" server has the option "stick to the top" enabled? Then the issue 1 won't be seen...

**Comment by math-GH** [comment]
I cannot help you when you do not answer my questions:

- did you check your settings in section "Display"? Which settings do you have?
- can you reproduce your issue on demo.freshrss.org?

**Comment by rg-software** [comment]
Yes, I understand. Just an unexpectedly tough day. Let's try again.

1) The checkbox `Stick the article to the top when opened` is **not** set in my case. I see what happens when I check it, so let's consider what FreshRSS is supposed to show when it is unchecked.
2) On the demo site this checkbox is apparently set, so it's a different case.

I tried Firefox and Safari, and I see the same behavior. So I was wondering how the system is supposed to work when the `Stick the article to the top` is unchecked?

My understanding is this.

Suppose I have two adjacent unread articles. I click on the first one to unfold it. Then I fold it back. Then I click on the second one and unfold it. This scenario works perfectly: When I collapse/uncollapse an article, the page doesn't move anywhere: I am exactly in the same place where I started.

Now, suppose that I uncollapse the first article, scroll it, then uncollapse the second article _without_ collapsing the first one. In this case the system collapses the first article automatically and uncollapses the second one. At this moment the browser navigates to some other place I do not expect it to be. What I expect is to see the second article uncollapsed right under the mouse cursor with the cursor still over its title.

So I was wondering how this sequence looks on your end. Sorry for longish explanations.

**Comment by Frenzie** [comment]
That seems to be "expected" but modern CSS has a few tricks that might help with that.

Edit: no, it doesn't seem like scroll-snap* stuff would.

**Comment by rg-software** [comment]
If you see the same as I see, then the `Stick the article to the top when opened` option seems compulsory. Maybe the lesser evil is keeping the manually uncollapsed articles uncollapsed. Personally I find that "auto-collapse" somewhat nice (if it works), but not really a killer feature. 

The "mark all" button on the toolbar is _the_ killer feature (in a bad way) that can ruin a day... not sure if anyone really needs  that thing right in the middle of the screen and with no confirmation box. But it's offtopic.

**Comment by Frenzie** [comment]
We'll have to ask someone else, presumably @Alkarex or @marienfressinaud 

It doesn't look to me like something that would've ever behaved any differently.

**Comment by rg-software** [comment]
I find it curious that it seemingly works fine on a mobile device, so it manages to collapse/uncollapse articles without losing the current focus.

**Comment by math-GH** [comment]
Can we summarize, that (1) is not a bug?

**Comment by rg-software** [comment]
Well, it's not for me to judge. Personally I think it is, because (a) it behaves differently on desktop and smartphone (it can't be intended); (b) it causes navigation to some unpredictable (undefined) place on a page.

**Comment by Alkarex** [comment]
Is this an accurate summary? When disabling the "stick" option (enabled by default), and when an article is expanded, opening another article makes the top of the newly opened article to move (potentially out of view)?
If yes, an idea could be to disable the auto-collapse of articles.

**Comment by Alkarex** [comment]
@rg-software Could you please give a try to this patch experiment? https://github.com/FreshRSS/FreshRSS/pull/4071

**Comment by rg-software** [comment]
Yes! 
I can confirm it works -- it doesn't collapse the article anymore, but it doesn't jump around the page either. At least for me this is a clear improvement. Thank you!
